server/routes/api/__test__/user.unit.test.js

```diff
@@ -1,7 +1,127 @@
 const request = require('supertest');
 const app = require('../../../../app');
+const User = require('../../../models/user');
+
 describe('User Routes', () => {
-  test('GET method', done => {
-    done();
+  let userID;
+  afterAll(() => {
+    if (userID) {
+      User.findByIdAndDelete({ _id: userID }).exec();
+    }
+  });
+
+  describe('GET /user', () => {
+    test('GET method', done => {
+      done();
+    });
+  });
+
+  describe('POST /user', () => {
+    test('POST method with no payload', () => {
+      return request(app)
+        .post('/api/user')
+        .type('form')
+        .then(res => {
+          expect(res.statusCode).toEqual(400);
+          expect(Object.keys(res.body.error.errors).length).toEqual(2);
+          expect(Object.keys(res.body.error.errors)).toEqual(
+            expect.arrayContaining(['email', 'password'])
+          );
+          expect(res.body.error.errors.email.message).toBe(
+            'You must provide an email address.'
+          );
+          expect(res.body.error.errors.password.message).toBe(
+            'You must provide a password.'
+          );
+        });
+    });
+
+    test('POST method with no email', () => {
+      return request(app)
+        .post('/api/user')
+        .type('form')
+        .send('password=test')
+        .then(res => {
+          expect(res.statusCode).toEqual(400);
+          expect(Object.keys(res.body.error.errors).length).toEqual(1);
+          expect(Object.keys(res.body.error.errors)).toEqual(
+            expect.arrayContaining(['email'])
+          );
+          expect(res.body.error.errors.email.message).toBe(
+            'You must provide an email address.'
+          );
+        });
+    });
+
+    test('POST method with no password', () => {
+      return request(app)
+        .post('/api/user')
+        .type('form')
+        .send('email=example@test.com')
+        .then(res => {
+          expect(res.statusCode).toEqual(400);
+          expect(Object.keys(res.body.error.errors).length).toEqual(1);
+          expect(Object.keys(res.body.error.errors)).toEqual(
+            expect.arrayContaining(['password'])
+          );
```

Comment:
The above tests seem to be a tad granular. Since we know that we want the error object to only have one key, and we know what that key is going to be, we can replace lines 63-66 with `expect(Object.keys(errorObject)).toEqual(['password'])`

Reply:
Refactored for all tests with similar error object testing.